### PR TITLE
[WIP] leptonica: update to 1.83.1, adopt.

### PR DESCRIPTION
**Pull request by Eloitor**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-musl)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - i686

### CODE REVIEW

**Comment by Eloitor** [comment]
What should I do with this?

```
ccextractor-0.93_1: broken, unresolvable shlib `liblept.so.5'
tesseract-ocr-4.1.1_7: broken, unresolvable shlib `liblept.so.5'
Transaction aborted due to unresolved shlibs.
Failed to install 'leptonica' and 'ccextractor-0.93_1'
```

**Comment by icp1994** [comment]
From the changelog
> Increase the .so number from 5.4.0 to 6.0.0.

As per my understanding, you need to update common/shlibs file accordingly and rebuild the packages from `xrevshlib leptonica`

File: srcpkgs/leptonica/template [hunk 1]
```diff
@@ -1,18 +1,18 @@
 # Template file for 'leptonica'
 pkgname=leptonica
-version=1.82.0
+version=1.83.1
 revision=1
-build_style=gnu-configure
+build_style=cmake
```
Comment (icp1994) [inline srcpkgs/leptonica/template:5]:
```suggestion
build_style=cmake
configure_args="-DBUILD_PROG=ON -DBUILD_SHARED_LIBS=ON"
```

File: srcpkgs/leptonica/template [hunk 2]
```diff
@@ -27,7 +27,6 @@ leptonica-devel_package() {
 		vmove usr/include/leptonica
 		vmove usr/lib/cmake
 		vmove usr/lib/pkgconfig
-		vmove "usr/lib/*.so"
```
Comment (icp1994) [inline srcpkgs/leptonica/template:30]: revert this change

**Comment by Eloitor** [comment]
maybe I need to update tesseract-ocr to fix the build problem that I have now.

**Comment by icp1994** [comment]
https://github.com/tesseract-ocr/tesseract/issues/3815#issuecomment-1123204962 yeah looks like it

**Comment by Eloitor** [comment]
The build of tesseract-ocr fails with:

```
  CXX      src/lstm/libtesseract_lstm_la-input.lo
  CXX      src/lstm/libtesseract_lstm_la-lstmrecognizer.lo
  CXX      src/lstm/libtesseract_lstm_la-lstm.lo
mv: cannot stat 'src/lstm/.deps/libtesseract_lstm_la-convolve.Tpo': No such file or directory
make[1]: *** [Makefile:6458: src/lstm/libtesseract_lstm_la-convolve.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
  CXX      src/lstm/libtesseract_lstm_la-maxpool.lo
  CXX      src/lstm/libtesseract_lstm_la-network.lo
  CXX      src/lstm/libtesseract_lstm_la-networkio.lo
  CXX      src/lstm/libtesseract_lstm_la-parallel.lo
  CXX      src/lstm/libtesseract_lstm_la-plumbing.lo
  CXX      src/lstm/libtesseract_lstm_la-recodebeam.lo
  CXX      src/lstm/libtesseract_lstm_la-reconfig.lo
  CXX      src/lstm/libtesseract_lstm_la-reversed.lo
  CXX      src/lstm/libtesseract_lstm_la-series.lo
  CXX      src/lstm/libtesseract_lstm_la-stridemap.lo
  CXX      src/lstm/libtesseract_lstm_la-tfnetwork.lo
  CXX      src/lstm/libtesseract_lstm_la-weightmatrix.lo
  CXXLD    libtesseract_native.la
  CXXLD    libtesseract_avx.la
  CXXLD    libtesseract_avx2.la
  CXXLD    libtesseract_avx512.la
  CXXLD    libtesseract_fma.la
  CXXLD    libtesseract_sse.la
  CXXLD    libtesseract_training.la
  CXXLD    libtesseract_ccutil.la
make[1]: Leaving directory '/builddir/tesseract-ocr-5.3.0'
make: *** [Makefile:8197: all-recursive] Error 1
make: *** Waiting for unfinished jobs....
=> ERROR: tesseract-ocr-5.3.0_1: do_build: '${make_cmd} ${makejobs} ${make_build_args} ${make_build_target}' exited with 2
=> ERROR:   in do_build() at common/build-style/gnu-configure.sh:15
```
